snuba/clickhouse/astquery.py

```diff
@@ -118,23 +120,37 @@ def _format_query_impl(self) -> str:
         if self.__limit is not None:
             limit_clause = f"LIMIT {self.__limit} OFFSET {self.__offset}"
 
-        self.__formatted_query = " ".join(
-            [
-                c
-                for c in [
-                    select_clause,
-                    from_clause,
-                    array_join_clause,
-                    prewhere_clause,
-                    where_clause,
-                    group_clause,
-                    having_clause,
-                    order_clause,
-                    limitby_clause,
-                    limit_clause,
-                ]
-                if c
+        self.__sql_data_list = [
```

Comment:
I've applied the empty clause culling to the sql_data dict. The empty strings don't provide much value.

Reply:
I need to fix my tests, of course.